src/main/java/io/confluent/connect/jdbc/dialect/MySqlDatabaseDialect.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.connect.jdbc.dialect;
+
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.connect.data.Date;
+import org.apache.kafka.connect.data.Decimal;
+import org.apache.kafka.connect.data.Time;
+import org.apache.kafka.connect.data.Timestamp;
+
+import java.util.Collection;
+
+import io.confluent.connect.jdbc.dialect.DatabaseDialectProvider.SubprotocolBasedProvider;
+import io.confluent.connect.jdbc.sink.metadata.SinkRecordField;
+import io.confluent.connect.jdbc.util.ColumnId;
+import io.confluent.connect.jdbc.util.ExpressionBuilder;
+import io.confluent.connect.jdbc.util.ExpressionBuilder.Transform;
+import io.confluent.connect.jdbc.util.IdentifierRules;
+import io.confluent.connect.jdbc.util.TableId;
+
+/**
+ * A {@link DatabaseDialect} for MySQL.
+ */
+public class MySqlDatabaseDialect extends GenericDatabaseDialect {
+  /**
+   * The provider for {@link MySqlDatabaseDialect}.
+   */
+  public static class Provider extends SubprotocolBasedProvider {
+    public Provider() {
+      super(MySqlDatabaseDialect.class.getSimpleName(), "mariadb", "mysql");
+    }
+
+    @Override
+    public DatabaseDialect create(AbstractConfig config) {
+      return new MySqlDatabaseDialect(config);
+    }
+  }
+
+  /**
+   * Create a new dialect instance with the given connector configuration.
+   *
+   * @param config the connector configuration; may not be null
+   */
+  public MySqlDatabaseDialect(AbstractConfig config) {
+    super(config, new IdentifierRules(".", "`", "`"));
+  }
+
+  @Override
+  protected String getSqlType(SinkRecordField field) {
```

Comment:
This is one of the few methods that _must_ be implemented in order to use this dialect in the sink connector. It maps from the record's schema type to the DBMS-specific data type.

But also note how this whole class is actually pretty short!